course_discovery/populate_course_run_reporting_type_and_announcement.py

```diff
@@ -0,0 +1,114 @@
+import argparse
+import datetime
+from dateutil.parser import parse
+import json
+import logging
+import os
+
+import logging
+
+from edx_rest_api_client.client import EdxRestApiClient
+
+
+logger = logging.getLogger()
+
+
+class CatalogApiService(object):
+    """The service to interface with edX catalog API"""
+
+    def __init__(self, oauth_host, oauth_key, oauth_secret, api_url_root):
+        try:
+            self.access_token, expires = EdxRestApiClient.get_oauth_access_token(
+                '{root}/access_token'.format(root=oauth_host),
+                oauth_key,
+                oauth_secret, token_type='jwt'
+            )
+        except Exception:
+            logger.exception('No access token acquired through client_credential flow.')
+            raise
+
+        logger.info('Token retrieved: %s', access_token)
+
+        self.api_client = EdxRestApiClient(api_url_root, jwt=self.access_token)
+
+    def _get_resource_from_api(self, api_endpoint, page_size=None, **kwargs):
```

Comment:
This is not used. Remove it. It can be restored if it's actually needed in the future.